### PR TITLE
Declaration sorting of script members

### PR DESCRIPTION
**Pull request by honzapatCZ**
Its not perfect, handles
fields and auto generated properties
I believe there is no built-in way to sort normal properties as this information is absent from the resulting assembly
related: #301

### CODE REVIEW

**Comment by honzapatCZ** [comment]
After merge, dont close this nor the issue, as this needs more improvements, note that its still better than the current alphabetical ordering which is just useless

**Comment by stefnotch** [comment]
Ideally this would be a conditional setting, maybe one that can be enabled by switching on an editor option.
(Like mafiesto4 suggested on Discord)

**Comment by honzapatCZ** [comment]
> Ideally this would be a conditional setting, maybe one that can be enabled by switching on an editor option.
> (Like mafiesto4 suggested on Discord)

What about voting from users?. 
Cuzz like get this, currently you get the alphabet which you:
1) don't care, so you wont care about declaration order either
2) use order attributes to change the order however you want, which you can still do

**Comment by stefnotch** [comment]
I had those sort of things in mind when I suggested the setting 😉 If it proves to be really popular and we figure out how to make the code really bulletproof, it could become a default setting.

For now, as you noticed, it has issues with normal properties and I'm not quite sure if we should make something potentially unstable a default.

File: Source/Editor/CustomEditors/Editors/GenericEditor.cs [hunk 1]
```diff
@@ -173,16 +173,16 @@ public int CompareTo(object obj)
                             return string.Compare(Display.Group, other.Display.Group, StringComparison.InvariantCulture);
                     }
 
-                    // By declaration order
-                    if (Info.MetadataToken > other.Info.MetadataToken)
-                        return 1;
-                    else if (Info.MetadataToken < other.Info.MetadataToken)
-                        return -1;
-                    else
+                    if(Editor.Instance.Options.Options.General.ScritpMembersOrder == Options.GeneralOptions.MembersOrder.Declaration)
```
Comment (stefnotch) [inline Source/Editor/CustomEditors/Editors/GenericEditor.cs:176]: Typo here, it should probably say `Script`

Comment (honzapatCZ) [reply Source/Editor/CustomEditors/Editors/GenericEditor.cs]: XD fixed

**Comment by mafiesto4** [comment]
Thanks!

**Pull request merged**